carousel/carousel.js

```diff
@@ -0,0 +1,228 @@
+        carouselRightButton.style = carouselOptions.buttonStyle;
+        carouselLeftButton.style = carouselOptions.buttonStyle;
+    }
+
+    carousel.addEventListener("keydown", (event) => {
+        if (event.key === "ArrowRight") {
+            slideCarouselItemsToRight(carousel);
+        }
+        else if (event.key === "ArrowLeft") {
+            slideCarouselItemsToLeft(carousel);
+        }
+    });
+
+    const carouselContentAndButtons = document.createElement("div");
+    carouselContentAndButtons.classList.add("carousel-content-and-buttons");
+    carouselContentAndButtons.append(carouselLeftButton, carouselContentHolder, carouselRightButton);
+
+    const carouselDotHolder = generateCarouselDotHolder(carouselItems.length);
+
+    if ("showDots" in carouselOptions &&
+        typeof carouselOptions.showDots === "boolean" &&
+        carouselOptions.showDots === false) {
+        carouselDotHolder.style.display = "none";
+    }
+
+    carousel.append(carouselTitle, carouselContentAndButtons, carouselDotHolder);
+
+    carousel.dataset.currentIndex = "0";
+    hideAdjacentCarouselItems(carousel);
+    updateLocationOfAdjacentCarouselItems(carousel);
+    updateCarouselDots(carousel);
+
+    carousel.addEventListener("transitionend", () => hideAdjacentCarouselItems(carousel, carouselContentHolder));
+
+    if ("autoplayTimeInMs" in carouselOptions &&
+        typeof carouselOptions.autoplayTimeInMs === "number") {
+        carousel.dataset.autoplayTimeInMs = carouselOptions.autoplayTimeInMs;
+    }
+    else {
+        carousel.dataset.autoplayTimeInMs = "6000";
+    }
+
+    resetCarouselAutoplay(carousel);
+
+    return carousel;
+}
+
+// Goes over all items with the carousel class and turns them to proper carousels.
+function generateCarousels() {
+    const carousels = Array.from(document.querySelectorAll('.carousel'));
+
+    for (let carousel of carousels) {
+        carousel.replaceWith(createCarousel(carousel.dataset.title, Array.from(carousel.children)));
+    }
+}
+
+function generateCarouselContentHolder(carouselContents) {
+    const carouselContentHolder = document.createElement("div");
+    carouselContentHolder.classList.add("carousel-content-holder");
+    carouselContentHolder.append(...carouselContents);
+
+    carouselContents.forEach((carouselItem) => {
+        carouselItem.classList.add("hidden");
+    });
+
+    return carouselContentHolder;
+}
+
+function generateCarouselDotHolder(amountOfDots) {
+    const carouselDotHolder = document.createElement("div");
+    carouselDotHolder.classList.add("carousel-dot-holder");
+
+    const carouselDot = document.createElement("div");
+    carouselDot.classList.add("carousel-dot");
+
+    for (let i = 0; i < amountOfDots; i++) {
+        carouselDotHolder.append(carouselDot.cloneNode());
+    }
+
+    return carouselDotHolder;
+}
+
+function updateCarouselDots(carousel) {
+    const currentIndexNumber = parseInt(carousel.dataset.currentIndex);
```

Comment:
Index is always a number in developers' head

```suggestion
    const currentIndex = parseInt(carousel.dataset.currentIndex);
```